python/lsst/ctrl/bps/parsl/sites/princeton.py

```diff
@@ -104,4 +104,4 @@ def get_address(self) -> str:
         interface, because the cluster nodes can't connect to the head node
         through the regular internet.
         """
-        return address_by_interface("ib0")
+        return address_by_interface("op0")
```

Comment:
This seems fragile if we have to change it a lot. Have you considered having an environment variable fallback?

Reply:
My feeling is that this shouldn't change a lot, but I may be off base about that. Do you have an example of env vars in use elsewhere that we might use here as an example?

One possibility I considered is making use of the `get_all_addresses` method within [parsl.addresses](https://github.com/Parsl/parsl/blob/6eae3547f46dcdb3a27e120b559bcbcad8c12ef2/parsl/addresses.py#L100), but I'm not sure how to use the information provided there to make the choice on the 'best' address to use.

Reply:
I've been in touch with Princeton IT regarding whether or not this variable is likely to change again in the future. To quote:
> this will not and cannot change. op0 was always the main high speed interface for tiger2 cluster
I think therefore that the fix we're making on this ticket is changing the Princeton site settings to what they should have been originally, and this is not likely to change again in the future. With that in mind, I'm inclined to leave this as-is and not add any user-specified fall-back. I.e., if this _does_ break in the future, it's probably something we want to fail fast and fix, rather than be hidden behind any kind of fall back procedure.

Reply:
On further discussion with a different member of staff within Princeton IT, it appears that this device interface may indeed change back to `ib0` or `ibN` at some future date, likely when the clusters are eventually upgraded. With that in mind, I've modified this ticket to make use of the same back-end within `parsl` (the `psutil` function `net_if_addrs`) to list all possible network interfaces and match only to those starting with either `ib*` or `op*`. I hope this will make us robust to any future configuration updates, removing the need to fix this issue if/when it crops up again.

@PaulPrice, do these new changes look okay to you?